common/src/test/java/io/druid/collections/StupidPoolTest.java

```diff
@@ -72,10 +70,24 @@ public void testExceptionInResourceHolderGet() throws IOException
     resourceHolderObj.get();
   }
 
-  @Test
-  public void testFinalizeInResourceHolder()
+  @Test(timeout = 60_000)
+  public void testResourceHandlerClearedByJVM()
+  {
+    if (System.getProperty("java.version").startsWith("1.7")) {
+      // This test is unreliable on Java 7, probably GC is not triggered by System.gc()
+      return;
+    }
+    createDanglingObjectHandler();
+    long poolSize = poolOfString.poolSize();
+    // Wait until dangling object string is returned to the pool
+    while (poolOfString.poolSize() == poolSize) {
+      System.gc();
+    }
+    Assert.assertEquals(poolSize + 1, poolOfString.poolSize());
+  }
+
+  private void createDanglingObjectHandler()
```

Comment:
I prefer to keep the method because it documents the code. If inlined, need to add a comment like "Here we create a dangling object handler by intentionally not storing it in a local variable".